### ISSUE TITLE
Units in moment maps wrong

### ISSUE THREAD

**Issue by SoFiA-Admin** [SoFiA-Admin/SoFiA#107]
There appear to be several issues with the units that appear in moment maps created by SoFiA. The global moment 0 and moment 1 maps appear to be correct and use the native units of the data cube, e.g. Jy/beam and Hz for flux density and frequency in the example I looked at. However, when creating cubelets for each source, the generated moment 0, 1 and 2 maps have units of Jy/beam km/s, km/s and km/s, respectively, so apparently the original frequency values are somehow converted to velocity. At the same time:
- The velocity values in the moment 1 map are nonsensical (almost 300,000 km/s for a source near 1420 MHz), while the values in the moment 0 and 2 maps look reasonable at first glance.
- The `DATAMIN` and `DATAMAX` keywords in the moment 2 map are wrong (close to 300,000 km/s; similar to the wrong velocity values in moment 1).
- The following warning message appears in SoFiA: `WARNING: Cannot convert axis3 coordinates to frequency. Will ignore the effect of CELLSCAL = 1/F`.

These issues need to be fixed and harmonised across SoFiA. In addition, we need to agree on whether we want native units or km/s for the third axis. Alternatively, we could let the users chose which units they want.

**Comment by SoFiA-Admin** [comment]
I just checked the code, and moment maps of cubelets appear to be created in `cubelets.py`. However, that file is currently a total nightmare due to inconsistent whitespace usage (see code example below where `»` marks a tab and ` ` or `.` marks a space), so the whitespace issue needs to be fixed first. I would generally recommend to introduce some kind of token to mark the end of programming blocks in our SoFiA Python code, e.g. `#end`, to avoid potential disasters caused by Python’s lack of an end-of-block delimiter.

![sofia_cubelets](https://cloud.githubusercontent.com/assets/6572706/19062862/0aef2040-89ed-11e6-9e7f-a506cacf8915.png)

**Comment by bwinkel** [comment]
> I would generally recommend to introduce some kind of token to mark the end of programming blocks in our SoFiA Python code, e.g. #end, to avoid potential disasters caused by Python’s lack of an end-of-block delimiter.

I wouldn't do this! Usually, if one follows [PEP8](https://www.python.org/dev/peps/pep-0008/) this isn't necessary. For many text editors there are good pep8 and linter plugins that indicate if one is not following the code style principles. Regarding white space, almost all python development teams enforce using only spaces (never tabs) with a width of 4 spaces per indentation level.

By-the-way, if we want to reach astropy affiliate package status at some point, we will need to strictly follow PEP8 anyway (among other [principles](http://www.astropy.org/affiliated/#affiliated-instructions)). I fear, this would be a very large work item, though.

**Comment by SoFiA-Admin** [comment]
> Usually, if one follows PEP8 this isn't necessary.

Yes, but unfortunately the world is not perfect, and the actual SoFiA code base isn’t even close to PEP8, so we have to accept that as a reality. Besides, the lack of end-of-block delimiters in Python is essentially a fatal design error in the Python language (see [this humorous blog post](http://www.hackerdashery.com/2006/09/close-block-delimiter-symbols.html) for the reasoning) that creates entirely unnecessary issues like the one with our `cubelets.py` module, in particular when multiple programmers collaborate on a single code base.

Furthermore, there are convincing arguments in favour of using tabs rather than spaces for indentation (e.g., see [this blog post](https://lea.verou.me/2012/01/why-tabs-are-clearly-superior/) for a summary). I used spaces myself in the past, but the arguments in favour of tabs are overwhelming, so I switched to tabs some time ago.

> By-the-way, if we want to reach astropy affiliate package status at some point, we will need to strictly follow PEP8 anyway (among other principles).

I don’t believe that we should even strive for that. Our main goal should remain to develop an accurate and reliable source finding pipeline for large HI data cubes.

**Comment by bwinkel** [comment]
I'm not very dogmatic about tabs or spaces, but I think it's important not to mix the two in a collaborative project, because it can create a git nightmare. (There are tools to auto-convert when checking-in/out, but the repository should stick to one style.)

Regarding pep8, I've spotted a nice little [tool](https://pypi.python.org/pypi/PythonTidy/) that can re-format python code to adhere to pep8. Usually, only little post-processing is necessary. I think, it's really worth it. I find my own code much easier to read since I switched to pep8.
